roles/ceph-osd/tasks/main.yml

```diff
@@ -53,6 +53,14 @@
   run_once: true
   delegate_to: "{{ groups['ceph_monitors'][0] }}"
 
+- name: create openstack vm pool if nova's backend is ceph
+  ceph_pool:
+    pool_name: "{{ ceph.vm_pool_name }}"
+  register: pool_output
+  run_once: true
+  delegate_to: "{{ groups['ceph_monitors'][0] }}"
+  when: nova.backend_ceph|bool
+
```

Comment:
Thinking about this more. The creation and definition of the VM pool should live with the consumer, Nova. Nova role can decide whether or not to use ceph, and create the pool it wants to use. The ceph roles should be ignorant to the consumers in that fashion.